### PR TITLE
Fix attribute name collisions in docstring

### PR DESCRIPTION
**Pull request by toslunar**
The attribute `gamma` in https://docs.chainer.org/en/v5.0.0b2/reference/generated/chainer.links.BatchNormalization.html#chainer.links.BatchNormalization links to the document of `BatchRenormalization`.  Fix this kind of errors.

cf. #3068

### CODE REVIEW

**Comment by toslunar** [comment]
Now we can revert some changes in #3068.
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainer/_sphinx_ext.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        except AttributeError:
+            q = klass.__name__
+        return '~%s.%s' % (q, attr_name)
+    return attr_name
+
+
+def setup(app):
+    app.setup_extension('sphinx.ext.napoleon')
+
+    def _parse_attributes_section(self, section):
```
Comment (kmaehashi) [inline chainer/_sphinx_ext.py]: I haven't looked into the details yet, but could you write down some comment about the difference to the original `_parse_attributes_section` function?

File: chainer/_sphinx_ext.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import sphinx
```
Comment (kmaehashi) [inline chainer/_sphinx_ext.py]: I think it's better to place this file under `docs/source` with name like `_napoleon_patch.py` or sth.
Then `extensions = ['sphinx.ext.autodoc', ... , '_napoleon_patch']` in `conf.py` should work.

**Comment by kmaehashi** [comment]
Also to-be-ported to CuPy.

**Comment by toslunar** [comment]
I submitted the same fix to the official repo since the change might be acceptable to others.
sphinx-doc/sphinx#5129

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM! Test failures are not related to this PR.

**Pull request merged**